robosuite/controllers/README.md

```diff
@@ -0,0 +1,43 @@
+
+Below, a brief overview and description of each subset of controller parameters are shown:
+
+#### General Controller Settings  
+* `'type'`: Type of controller to control. Can be `'EE_POS_ORI'`, `'EE_POS'`, `'EE_IK'`, `'JOINT_IMP'`, `'JOINT_VEL'`, or `'JOINT_TORQUE'`
+* `'interpolation'`: If not `null`, specified type of interpolation to use between desired actions. Currently only `'linear'` is supported. 
+* `'{...}_limits'`: Limits for that specific controller. E.g.: for a `'JOINT_IMP'`, the relevant limits are its joint positions, `'qpos_limits'` . Can be either a 2-element list (same min/max limits across entire relevant space), or a list of lists (specific limits for each component)
+* `'{input,output}_{min,max}'`: Scaling ranges for mapping action space inputs into controller inputs. Settings these limits will automatically clip the action space input to be within the `'input_{min,max}'` before mapping the requested value into the specified `'output_{min,max}'` range. Can be either a scalar (same limits across entire action space), or a list (specific limits for each action component)
+* `'kp', 'kv'`: Where relevant, specifies the positional / velocity gain for the controller. Can be either be a scalar (same value for all robot joints), or a list (specific values for each joint)
+* `'damping'`: Where relevant, specifies the damping constant for the controller.
+* `'control_delta'`: Only relevant for `'EE_POS_ORI'` or `'EE_POS'` controllers. `true` interprets input actions as delta values from the current robot end effector position. Otherwise, assumed to be absolute (global) values
+* `'uncouple_pos_ori'`: Only relevant for `'EE_POS_ORI'`. `true` decouples the desired position and orientation when executing the controller
+
+## Using a Custom Controller Configuration
+A custom controller other than the environment defaults (which are normally are the default `JOINT_VEL` configurations) can be used by simply creating a new config (`.json`) file with the relevant parameters as specified above. All robosuite environments have an optional `controller_config` argument that can be used to pass in specific controller settings. Note that this is expected to be a `dict`, so the new configuration must be read in and parsed as a `dict` before passing it during the environment `robosuite.make(...)` call. A brief example script showing how to import a custom controller configuration is shown below.
+
+```python
+import robosuite as suite
+import json
+
+# Path to config file
+controller_fpath = `/your/custom/config/filepath/here/filename.json`
+
+# Import the file as a dict
+with open(controller_fpath) as f:
+	config = json.load(f)
```

Comment:
can you fix this spacing to be 4 spaces?

Reply:
done